src/game.cpp

```diff
@@ -7968,9 +7984,89 @@ bool pet_menu(monster *z)
 
         return true;
     }
+    if ( inject == choice ) {
+        auto filter = [] (const item &it) {
+            return it.has_flag("MUTAGEN_CEPHALOPOD") ||
+                   it.has_flag("PURIFIER");
```

Comment:
This filter (and the check below) don't properly check for containment.
As a result the only way to inject mutagen is to spawn it in the inventory without a flask, which is only possible with debug menu.

Also, it probably shouldn't be possible to inject purifier into "pure" dog.